src/wasi.rs

```diff
@@ -9,15 +9,11 @@
 //! Implementation for WASI
 use crate::Error;
 use core::num::NonZeroU32;
-use wasi::random_get;
+use wasi::wasi_snapshot_preview1::random_get;
```

Comment:
Why not just use `wasi::random_get`?

Reply:
Because the new [`Errno`](https://docs.rs/wasi/latest/wasi/struct.Errno.html) from wasi, which replaced `Error`, can actually hold 0. `Errno::as_raw` returns `u16`, which may be zero, thus invalidating the previous code calling `NonZeroU32::new_unchecked`. However handling zero ourselves ensures we can call `NonZeroU32::new_unchecked` again.

Reply:
But it won't actually be zero if the function returns `Err`, so it seems fine.

Reply:
It means that soundness of our code relies on an implementation detail of `wasi`. I am not sure we can rely on it, even though it's a very reasonable assumption.

Reply:
> It would mean that soundness of our code relies on an implementation detail of `wasi`. I am not sure we can rely on it, even though it's a very reasonable assumption.

This is the exact reason I decided to switch.